### PR TITLE
xtask: add coverage command

### PR DESCRIPTION
**Pull request by davidhewitt**
This is a bit of an experiment to add a `cargo xtask coverage` alias.

Ultimately this will re-implement the coverage recipe in ci. I also was playing around with adding coverage from out examples and python tests. It doesn't seem to be working quite yet; needs upstream support in `cargo-llvm-cov` which I'm still trying to figure out.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -119,7 +119,8 @@ harness = false
 members = [
     "pyo3-macros",
     "pyo3-macros-backend",
-    "examples"
+    "examples",
+    "xtask"
```
Comment (mejrs) [inline Cargo.toml:123]: We don't necessarily need to name this "xtask". I don't have a motivation for changing it. But we could.

File: xtask/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+[package]
+name = "xtask"
+version = "0.1.0"
+edition = "2021"
+
+[dependencies]
+anyhow = "1.0.51"
+clap = { version = "3.0.0-rc.7", features = ["derive"] }
```
Comment (mejrs) [inline xtask/Cargo.toml:8]: Does this add `clap` to pyo3's dependency tree? If so can we avoid that?

**Review by mejrs — APPROVED** [review]
Thanks :)

I've played around a bit with this "xtask' approach on a project and I like it. I'd be happy to see this go forward. And to help implement it ❤️

File: xtask/Cargo.toml (hunk 2 above)
Comment (davidhewitt) [reply xtask/Cargo.toml:8]: I don't believe it does, no. This is an excerpt of `Cargo.lock` from a test project built against this branch:

```
[[package]]
name = "pyo3"
version = "0.15.1"
dependencies = [
 "cfg-if",
 "indoc",
 "libc",
 "num-bigint",
 "parking_lot",
 "paste",
 "pyo3-build-config",
 "pyo3-macros",
 "unindent",
]
```

So `clap` hasn't come through as a dependency here.

File: Cargo.toml (hunk 1 above)
Comment (davidhewitt) [reply Cargo.toml:123]: Yeah, I don't personally _love_ the name xtask but I do like that it's a documented "approach" (https://github.com/matklad/cargo-xtask), so without good motivation I don't feel a need to change it.

For example I'd personally enjoy the alias being called `cargo run-script`. There's nothing stopping us adding a couple extra aliases that all do the same later, if we wanted ;)

**Comment by davidhewitt** [comment]
Great! I've tidied this up a bit and added `cargo xtask test-py`, which now gets used in CI instead of `make test_py`. 

If CI passes I'll merge this now; I'll follow up on the coverage command improvements later once the `cargo-llvm-cov` support is finished, and @mejrs please do feel welcome to add to and expand to the set of commands implemented here to replace the Makefile :)

**Pull request merged**